src/viam/ur/module/ur_arm.cpp

```diff
@@ -215,6 +226,20 @@ URArm::URArm(Model model, const Dependencies& deps, const ResourceConfig& cfg) :
         throw std::runtime_error("couldn't connect to dashboard");
     }
 
+    // verify that the connected arm is the same model as the configured module
+    {
+        std::string conn_model_type{};
+        if (!current_state_->dashboard->commandGetRobotModel(conn_model_type)) {
+            throw std::runtime_error("failed to get model info of connected arm");
+        }
+
+        std::string cfg_model_type = to_robot_type_string(model_.to_string());
+        if (cfg_model_type != conn_model_type) {
+            const auto err = "configured model type `" + cfg_model_type + "` does not match connected arm `" + conn_model_type + "`";
+            throw std::runtime_error(err);
+        }
+    }
```

Comment:
scoped so we don't carry the strings longer than necessary

Reply:
Good to consider the resource implications, but here I wouldn't bother. It is just the constructor, which won't ever be on a fast path, and we can assume basically no memory pressure compared to the size of these strings. And I'd expect the compiler's lifetime analysis to figure it out anyway.